proxystore/connectors/dim/margo.py

```diff
@@ -221,7 +236,7 @@ def config(self) -> dict[str, Any]:
         the connector object.
         """
         return {
-            'interface': self.interface,
+            'addr': self.host,
```

Comment:
Here we need to be careful to return the *user* provided configuration and not the machine-specific configuration (e.g., we don't want to have the config be this specific machine's IP address).

Something like saving the initial state of arguments at the top of `__init__` before resolving the specific values.

This applies to all three implementations.